### ISSUE TITLE
Relative links

### ISSUE THREAD

**Issue by 0xCMP** [4Catalyzer/found#120]
Relative link navigate to the right url, but it seems that found does not know what route it is supposed to render if the url pushed was relative.

**Comment by taion** [comment]
There's no real support for relative links right now. I have no objection to adding them, but they're not intended to work right now.

**Comment by 0xCMP** [comment]
So I'm thinking of making the BaseLink component change it's URL to a non-relative one since it has the current location and the relative url.

I'd do a test for it being a relative url by checking for that it starts with "../" and "./" for now. I have cases where it can be like "1/" which should probably work too, but I don't want to make the check too complicated or break existing links.

I want to know if you have a preference for how I should check for relative urls or create the url using the existing location and relative url.

As of right now I found this to take the current location and apply a relative url to it: 

https://github.com/medialize/URI.js/blob/61cd727e65b8103110c6e207188e2250f5677213/src/URI.js#L2112-L2160

Example:

```javascript
var uri = new URI("../../../path");
// make path absolute
var relUri = uri.absoluteTo("/relative/sub/foo/sub/file"); // returns a new URI instance
// relUri == "/relative/path"
```

**Comment by taion** [comment]
It's a little trickier than that.

See e.g. https://github.com/ReactTraining/react-router/issues/2172#issuecomment-196081972.

Relative links should be relative to the route context, not to the current URL. That way you can put relative links on a top-level nav element, and have them still work on child routes.

**Comment by 0xCMP** [comment]
Oh, so the base link in a higher route level gets that route's location not the current URL?

**Comment by taion** [comment]
If you don't do that, then things would be weird, right?

**Comment by 0xCMP** [comment]
I thought that because of how active states worked that the current `location` object would point to the current url (e.g. so that exact vs non-exact could work on higher level routes) and that if the `to` was relative I could convert that relative url passed in to me to a full url given where I am right now. That way the url can be matched correctly if `router.push` was called.

**Comment by taion** [comment]
Right, so it'd take some work to put the "route path" on context or whatever, but otherwise you just wouldn't be able to do relative links properly.

**Comment by caseybaggz** [comment]
Yikes, spent so long figuring out why my router wasn't displaying content and it was all because of relative links... 😭

**Comment by hgossler** [comment]
It would be very helpful if the readme mentioned that links must all be absolute paths.
